comp/core/autodiscovery/scheduler/controller.go

```diff
@@ -0,0 +1,187 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package scheduler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+// MaxRetries is the maximum number of retries for a failed task (TODO: implement retries on failure)
+// MaxRetries = 5
+)
```

Comment:
Question

I see you are using workqueue, do we still need this TODO item?

Reply:
Because retry is still not implemented. The status must be checked [here](scheduler.Schedule(([]integration.Config{*desiredConfigState.config})) // TODO: check status of action) before retry. Currently, schedule/unschedule call doesn't return status yet.